volumes/miovision/sql/update-mio_volumes15min_mvt_partitioning.sql

```diff
@@ -0,0 +1,155 @@
+--DROP TABLE mio_staging.volumes_15min_mvt;
+CREATE TABLE mio_staging.volumes_15min_mvt (
+    --create staging table based on original.
+    LIKE miovision_api.volumes_15min_mvt INCLUDING DEFAULTS,
+    --change primary key to include datetime_bin for partioning
+    CONSTRAINT volumes_15min_mvt_intersection_uid_datetime_bin_classificat_pk
+    PRIMARY KEY (intersection_uid, datetime_bin, classification_uid, leg, movement_uid)
+)
+PARTITION BY RANGE (datetime_bin);
+
+COMMENT ON COLUMN mio_staging.volumes_15min_mvt.leg
+IS 'Leg of approach or entry point ';
+
+-- Index: volumes_15min_mvt_classification_uid_idx
+-- DROP INDEX IF EXISTS mio_staging.volumes_15min_mvt_classification_uid_idx;
+CREATE INDEX IF NOT EXISTS volumes_15min_mvt_classification_uid_idx
+ON mio_staging.volumes_15min_mvt USING btree(classification_uid ASC nulls last);
+
+-- Index: volumes_15min_mvt_datetime_bin_idx
+-- DROP INDEX IF EXISTS mio_staging.volumes_15min_mvt_datetime_bin_idx;
+CREATE INDEX IF NOT EXISTS volumes_15min_mvt_datetime_bin_idx
+ON mio_staging.volumes_15min_mvt USING brin(datetime_bin);
+
+-- Index: volumes_15min_mvt_intersection_uid_idx
+-- DROP INDEX IF EXISTS mio_staging.volumes_15min_mvt_intersection_uid_idx;
+CREATE INDEX IF NOT EXISTS volumes_15min_mvt_intersection_uid_idx
+ON mio_staging.volumes_15min_mvt USING btree(intersection_uid ASC nulls last);
+
+-- Index: volumes_15min_mvt_leg_movement_uid_idx
+-- DROP INDEX IF EXISTS mio_staging.volumes_15min_mvt_leg_movement_uid_idx;
+CREATE INDEX IF NOT EXISTS volumes_15min_mvt_leg_movement_uid_idx
+ON mio_staging.volumes_15min_mvt USING btree(
+    leg COLLATE pg_catalog."default" ASC nulls last, movement_uid ASC nulls last
+);
+
+-- Index: volumes_15min_mvt_processed_idx
+-- DROP INDEX IF EXISTS mio_staging.volumes_15min_mvt_processed_idx;
+CREATE INDEX IF NOT EXISTS volumes_15min_mvt_processed_idx
+ON mio_staging.volumes_15min_mvt USING btree(processed ASC nulls last)
+WHERE processed IS NULL;
+
+-- Index: volumes_15min_mvt_volume_15min_mvt_uid_idx
+-- DROP INDEX IF EXISTS mio_staging.volumes_15min_mvt_volume_15min_mvt_uid_idx;
+CREATE INDEX IF NOT EXISTS volumes_15min_mvt_volume_15min_mvt_uid_idx
+ON mio_staging.volumes_15min_mvt USING btree(volume_15min_mvt_uid ASC nulls last);
+    
+--create partitions
+--note this partition create function has been altered since.
+DO $do$
+DECLARE
+	yyyy TEXT;
+BEGIN
+	FOR yyyy IN 2019..2023 LOOP
+        EXECUTE FORMAT($$SELECT gwolofs.create_yyyy_partition(
```

Comment:
should this be another schema?

Reply:
This is fine, it was just the temp schema of the function at the I ran it.